single_kernel_mongo/managers/backups.py

```diff
@@ -0,0 +1,570 @@
+                    backup_status = "finished"
+                else:
+                    # display reason for failure if available
+                    backup_status = "failed: " + backup.get("error", "N/A")
+            if backup["status"] not in ["error", "done"]:
+                backup_status = "in progress"
+            backup_list.append((backup["name"], backup["type"], backup_status))
+
+        # process in progress backups
+        running_backup = pbm_status["running"]
+        if running_backup.get("type", None) == "backup":
+            # backups are sorted in reverse order
+            last_reported_backup = backup_list[0]
+            # pbm will occasionally report backups that are currently running as failed, so it is
+            # necessary to correct the backup list in this case.
+            if last_reported_backup[0] == running_backup["name"]:
+                backup_list[0] = (
+                    last_reported_backup[0],
+                    last_reported_backup[1],
+                    "in progress",
+                )
+            else:
+                backup_list.append((running_backup["name"], "logical", "in progress"))
+
+        return self._format_backup_list(sorted(backup_list, key=lambda pair: pair[0]))
+
+    @retry(
+        stop=stop_after_attempt(BACKUP_RESTORE_MAX_ATTEMPTS),
+        retry=retry_if_not_exception_type(RestoreError),
+        wait=wait_fixed(BACKUP_RESTORE_ATTEMPT_COOLDOWN),
+        reraise=True,
+        before_sleep=_backup_restore_retry_before_sleep,
+    )
+    def restore_backup(self, backup_id: str, remapping_pattern: str | None = None) -> None:
+        """Try to restore cluster a backup specified by backup id.
+
+        If PBM is resyncing, the function will retry to create backup
+        (up to  BACKUP_RESTORE_MAX_ATTEMPTS times) with BACKUP_RESTORE_ATTEMPT_COOLDOWN
+        time between attempts.
+
+        If PMB returen any other error, the function will raise RestoreError.
+        """
+        try:
+            remapping_pattern = remapping_pattern or self._remap_replicaset(backup_id)
+            remapping_args = ["--replset-remapping", remapping_pattern] if remapping_pattern else []
+            self.workload.run_bin_command(
+                "restore",
+                [backup_id] + remapping_args,
+                environment=self.environment,
+            )
+        except WorkloadExecError as e:
+            error_message = e.stdout
+            if "Resync" in e.stdout:
+                raise ResyncError
+
+            fail_message = f"Restore failed: {str(e)}"
+            if f"backup '{backup_id}' not found" in error_message:
+                fail_message = f"Restore failed: Backup id '{backup_id}' does not exist in list of backups, please check list-backups for the available backup_ids."
+
+            raise RestoreError(fail_message)
+
+    def get_status(self) -> StatusBase | None:
```

Comment:
Yes and as you can see now, it's different from the `healthy` you were asking earlier.
My idea (not formalized yet because statuses are for later) is that every manager will provide a `get_status` method so that then we can collect all of them and sort out the most important status.